### ISSUE TITLE
What is the best way to do evaluation after training?

### ISSUE THREAD

**Issue by yiminglin-ai** [ashleve/lightning-hydra-template#238]
Hi @ashleve,
Thanks for the work. I am wondering if what is the best practice on the post-training evaluation?
Say I want to eval the model on a different test set, what is the best way to do it with hydra? 
Many thanks.

**Comment by ashleve** [comment]
Yea so there are a couple of approaches, I would personally go with having a different main config and run file for train and evaluation. So something like this:
```
├── configs
│   ├── ...
│   ├── train.yaml
│   ├── evaluate.yaml
│ 
├── train.py
├── evaluate.py
```
I realize this doesn't go along well with the current approach of having `run.py` in root folder and `train.py` in `src`, we should probably redesign that, will think about it

**Comment by mariomeissner** [comment]
I agree that having an evaluate.py script would be useful. We often want to test our trained models on other datasets. It would be great if we could use the same configuration files we used for training to quickly pick up all the required configs, then override the datamodule we want to use for testing.

Some things need to be automatically overridden as well, such as output folder (results should be written into a file in the folder of the original model) and switch training mode to evaluation mode, etc.). 

I have attempted to do this by myself using the `mode` configs (a train mode and evaluation mode) but would be cool to see what you guys come up with.
